packages/api/src/router/github/github.ts

```diff
@@ -0,0 +1,349 @@
+import { createAppAuth } from "@octokit/auth-app";
+import { Octokit } from "@octokit/rest";
+import { TRPCError } from "@trpc/server";
+import _ from "lodash";
+import { isPresent } from "ts-is-present";
+import { z } from "zod";
+
+import { and, eq, inArray, takeFirstOrNull } from "@ctrlplane/db";
+import {
+  deployment,
+  githubConfigFile,
+  githubOrganization,
+  githubOrganizationInsert,
+  githubUser,
+  jobAgent,
+} from "@ctrlplane/db/schema";
+
+import { env } from "../../config";
+import { createTRPCRouter, protectedProcedure } from "../../trpc";
+import { createNewGithubOrganization } from "./create-github-org";
+
+const octokit =
+  env.GITHUB_BOT_APP_ID == null
+    ? null
+    : new Octokit({
+        authStrategy: createAppAuth,
+        auth: {
+          appId: env.GITHUB_BOT_APP_ID,
+          privateKey: env.GITHUB_BOT_PRIVATE_KEY,
+          clientId: env.GITHUB_BOT_CLIENT_ID,
+          clientSecret: env.GITHUB_BOT_CLIENT_SECRET,
+        },
+      });
+
+const getOctokitInstallation = (installationId: number) =>
+  new Octokit({
+    authStrategy: createAppAuth,
+    auth: {
+      appId: env.GITHUB_BOT_APP_ID,
+      privateKey: env.GITHUB_BOT_PRIVATE_KEY,
+      clientId: env.GITHUB_BOT_CLIENT_ID,
+      clientSecret: env.GITHUB_BOT_CLIENT_SECRET,
+      installationId,
+    },
+  });
+
+const getOctokit = () => {
+  if (octokit == null)
+    throw new TRPCError({
+      code: "INTERNAL_SERVER_ERROR",
+      message: "GitHub bot not configured",
+    });
+  return octokit;
+};
+
+const userRouter = createTRPCRouter({
+  byUserId: protectedProcedure
+    .input(z.string())
+    .query(({ ctx, input }) =>
+      ctx.db
+        .select()
+        .from(githubUser)
+        .where(eq(githubUser.userId, input))
+        .then(takeFirstOrNull),
+    ),
+
+  delete: protectedProcedure
+    .input(z.string())
+    .mutation(async ({ ctx, input }) =>
+      ctx.db.delete(githubUser).where(eq(githubUser.userId, input)),
+    ),
+
+  create: protectedProcedure
+    .input(
+      z.object({
+        userId: z.string(),
+        githubUserId: z.number(),
+        githubUsername: z.string(),
+      }),
+    )
+    .mutation(({ ctx, input }) =>
+      ctx.db
+        .insert(githubUser)
+        .values(input)
+        .returning()
+        .then((data) => data[0]),
```

Comment:
```suggestion
        .then(([data]) => data),
```

why do we need this? use takeFirst?